test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -1,55 +1,108 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+function Get-PipePath {
+    param (
+        $PipeName
+    )
+    if ($IsWindows) {
+        return "\\.\pipe\$PipeName"
+    }
+    "$([System.IO.Path]::GetTempPath())CoreFxPipe_$PipeName"
+}
+
+function Start-PSProcess {
+    param (
+        [switch] $WindowsPowerShell,
+        [array] $ArgumentList
+    )
+
+    $pwsh = if ($WindowsPowerShell) { "powershell" } else { "pwsh" }
+
+    $params = @{
+        FilePath = $pwsh
+        PassThru = $true
+        RedirectStandardOutput = "TestDrive:\$([System.IO.Path]::GetRandomFileName())"
+        RedirectStandardError = "TestDrive:\$([System.IO.Path]::GetRandomFileName())"
+    }
+
+    if ($ArgumentList) {
+        $params.ArgumentList = $ArgumentList
+    }
+
+    Start-Process @params
+}
+
 Describe "Enter-PSHostProcess tests" -Tag Feature {
     Context "By Process Id" {
-        BeforeAll {
-            $params = @{
-                FilePath = "pwsh"
-                PassThru = $true
-                RedirectStandardOutput = "TestDrive:\pwsh_out.log"
-                RedirectStandardError = "TestDrive:\pwsh_err.log"
-            }
-            $pwsh = Start-Process @params
-
-            if ($IsWindows) {
-                $params = @{
-                    FilePath = "powershell"
-                    PassThru = $true
-                    RedirectStandardOutput = "TestDrive:\powershell_out.log"
-                    RedirectStandardError = "TestDrive:\powershell_err.log"
-                }
-                $powershell = Start-Process @params
-            }
 
-        }
+        It "Can enter, exit, and re-enter another PSHost" {
+            # $pwsh = Start-PSProcess
```

Comment:
Why commented?

Reply:
You should probably just hold off on reviewing. This issue only repros in CI... so it'll be a lot of trial and error.

When I remove WIP, you can review for real.

Open to suggestions on how to fix the tests though 😅

Reply:
My suggestion was above - use "-file" parameter and put the temporary test script in file.